src/passes/literalExpressionEvaluator/literalExpressionEvaluator.ts

```diff
@@ -129,7 +126,80 @@ function evaluateBinaryLiteral(node: BinaryOperation): RationalLiteral | boolean
     evaluateLiteralExpression(node.vRightExpression),
   ];
   if (left === null || right === null) {
-    return null;
+    // In some cases a binary expression could be calculated at
+    // compile time, even when only one argument is a literal.
+    const notNullMember = left === null ? right : left;
+    if (notNullMember === null) {
+      return null;
+    }
+    if (typeof notNullMember === 'boolean') {
+      switch (node.operator) {
+        case '&&': // false & x = false
+          return notNullMember ? null : false;
+        case '||': // true | x = true
+          return notNullMember ? true : false;
+        default:
+          if (!['==', '!='].includes(node.operator)) {
+            throw new TranspileFailedError(
+              `Unexpected boolean x boolean operator ${node.operator}`,
+            );
+          }
+          return null;
+      }
+    } else if (typeof notNullMember !== 'boolean') {
+      const fraction = notNullMember.toString().split('/');
+      const is_zero = fraction[0] === '0';
+      const is_one = fraction[0] === fraction[1];
+      switch (node.operator) {
+        case '*': // 0*x = x*0 = 0
+          return is_zero ? new RationalLiteral(0n, 1n) : null;
+        case '**': // x**0 = 1   1**x = 1
+          if (right && is_zero) {
+            return new RationalLiteral(1n, 1n);
+          } else if (left && is_one) {
+            return new RationalLiteral(1n, 1n);
+          }
+          return null;
+        case '<<': // 0<<x = 0   x<<n(n>255) = 0
+          if (left && is_zero) {
+            return new RationalLiteral(0n, 1n);
+          } else if (right && notNullMember.greaterThan(new RationalLiteral(255n, 1n))) {
+            return new RationalLiteral(0n, 1n);
+          }
+          return null;
+        case '>>': // 0>>x = 0   1>>x = 0
+          if (left && is_zero) {
+            return new RationalLiteral(0n, 1n);
+          } else if (left && is_one) {
+            return new RationalLiteral(0n, 1n);
+          }
+          return null;
+        default: {
+          const otherOp = [
+            '/',
+            '%',
+            '+',
+            '-',
+            '>',
+            '<',
+            '>=',
+            '<=',
+            '==',
+            '!=',
+            '|',
+            '&',
+            '^',
+            '~',
+          ];
+          if (!otherOp.includes(node.operator)) {
+            throw new TranspileFailedError(`Unexpected number x number operator ${node.operator}`);
+          }
+          return null;
+        }
+      }
+    } else {
```

Comment:
Why would this `else` be needed? If the type of the `notNullMember` variable is a boolean it will enter the `if` statement and if it is not it will enter the `else if`, so what other case is there? Maybe consider using `else` instead of `else if` above.